optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/widgets/planner/PlannerDataObjectFieldEditorViewImpl.java

```diff
@@ -19,59 +19,57 @@
 import javax.enterprise.context.Dependent;
 import javax.inject.Inject;
 
-import com.google.gwt.core.client.GWT;
+import com.google.gwt.event.dom.client.ChangeEvent;
 import com.google.gwt.event.dom.client.ClickEvent;
-import com.google.gwt.event.logical.shared.ValueChangeEvent;
-import com.google.gwt.uibinder.client.UiBinder;
-import com.google.gwt.uibinder.client.UiField;
-import com.google.gwt.uibinder.client.UiHandler;
 import com.google.gwt.user.client.ui.Composite;
-import com.google.gwt.user.client.ui.Widget;
-import org.gwtbootstrap3.client.ui.CheckBox;
-import org.gwtbootstrap3.client.ui.Form;
 import org.gwtbootstrap3.client.ui.TextBox;
+import org.jboss.errai.common.client.dom.CheckboxInput;
+import org.jboss.errai.common.client.dom.Div;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
 
 @Dependent
+@Templated
 public class PlannerDataObjectFieldEditorViewImpl
         extends Composite
         implements PlannerDataObjectFieldEditorView {
 
-    interface Binder
-            extends UiBinder<Widget, PlannerDataObjectFieldEditorViewImpl> {
-
-    }
-
-    private static Binder uiBinder = GWT.create( Binder.class );
-
     private Presenter presenter;
 
-    @UiField
-    Form planningSolutionSettingsPanel;
+    @Inject
+    @DataField("planningSolutionSettingsPanelDiv")
+    Div planningSolutionSettingsPanelDiv;
 
-    @UiField
-    CheckBox valueRangeProviderCheckBox;
+    @Inject
+    @DataField("valueRangeProviderCheckBox")
+    CheckboxInput valueRangeProviderCheckBox;
 
-    @UiField
+    @Inject
+    @DataField("valueRangeProviderIdTextBox")
```

Comment:
I'm curious, what is the reason for keeping this field as a TextBox rather than a TextInput?